src/bloom.cpp

```diff
@@ -34,12 +34,12 @@ CBloomFilter::CBloomFilter(unsigned int nElements, double nFPRate, unsigned int
      * Again, we ignore filter parameters which will create a bloom filter with more hash functions than the protocol limits
      * See https://en.wikipedia.org/wiki/Bloom_filter for an explanation of these formulas
      */
-    isFull(false),
     isEmpty(false),
     nHashFuncs(min((unsigned int)(vData.size() * 8 / nElements * LN2), MAX_HASH_FUNCS)),
     nTweak(nTweakIn),
     nFlags(nFlagsIn)
 {
+    isFull = vData.size() == 0;
```

Comment:
@dgenr8: So it sounds like you favor the 'zero-size is a valid value, however needs to be properly checked for'-angle?  What I like about this is that it seems to be less code to change.

What do you think about the upper limits, though? If I deserialize, I can in principle make an almost 32MB-sized bloom filter. Should I just make resize it to zero in that case?

Of course, I can check with IsWithInSizeConstraints(..). However, I like to find a way to make the filter robust with as little calls as possible. Do you have a good idea here?